cmd/booster-bitswap/server.go

```diff
@@ -58,12 +64,26 @@ func (s *BitswapServer) Start(ctx context.Context, proxy *peer.AddrInfo) error {
 	if err != nil {
 		return err
 	}
-	bsopts := []server.Option{server.MaxOutstandingBytesPerPeer(1 << 20), server.WithPeerBlockRequestFilter(func(p peer.ID, c cid.Cid) bool {
-		fulfill, err := s.filter.FulfillRequest(p, c)
-		// peer request filter expects a true if the request should be fulfilled, so
-		// we only return true for requests that aren't filtered and have no errors
-		return fulfill && err == nil
-	})}
+	bsopts := []server.Option{
+		server.MaxOutstandingBytesPerPeer(1 << 20),
+		server.WithPeerBlockRequestFilter(func(p peer.ID, c cid.Cid) bool {
+			ss := filters.ServerState{}
+			// TODO: this really isn't ideal - we should find a way to get these values passed into the peer filter directly
+			if s.server != nil {
+				ss = filters.ServerState{
+					TotalRequestsInProgress:   s.server.TotalWants(),
+					RequestsInProgressForPeer: s.server.WantCountForPeer(p),
+				}
+			}
```

Comment:
I'm not sure that wants correspond to requests in progress. IIRC the want list is the list of blocks that a client wants from the server. If the server doesn't have the block, the want remains in the want list until it's removed by the client with a cancel message.

Reply:
I figured this one out I think and removed the need for the bitswap PR as a bonus!